src/Changelog/BumpAndCommitChangelogVersionViaKeepAChangelog.php

```diff
@@ -62,16 +61,15 @@ public function __invoke(
 
         ($this->checkoutBranch)($repositoryDirectory, $sourceBranch);
 
-        $changelogFile = sprintf('%s/%s', $repositoryDirectory, self::CHANGELOG_FILE);
+        $changelogFile = Str\format('%s/%s', $repositoryDirectory, self::CHANGELOG_FILE);
         $versionString = $version->fullReleaseName();
         $bumper        = new ChangelogBump($changelogFile);
-        $newVersion    = $bumper->$bumpType($versionString);
+        $newVersion    = Type\non_empty_string()->assert($bumper->$bumpType($versionString));
```

Comment:
IMO coerce? Or does `assert()` return anything?

Reply:
assert returns as well.

difference:

coerce: given `$value` of `mixed it will try to convert `$value` to `T` and return `T`, otherwise throws.
assert: given `$value` of `mixed` it will assert that `$value` is already `T` and return it, otherwise throws.

e.g: `bool()->coerce(1); // true`, `bool()->assert(1); // throws`.

Reply:
in this specific scenario, `coerce(45)` will return`"45"` ( since it can be converted to a non-empty-string ), `assert(45);` will throw  .

so you should use assert whenever possible.

Reply:
Excellent, good to know! :+1: